labgrid/protocol/commandprotocol.py

```diff
@@ -1,6 +1,87 @@
 import abc
 
 
+class CommandProcessProtocol(abc.ABC):
+    """Abstract class for a running command"""
+
+    @abc.abstractmethod
+    def read(self, size=-1, *, timeout=None):
+        """
+        Reads bytes from the process until either size bytes have been read,
+        timeout seconds have elapsed, or EOF is encountered. Returns as many
+        bytes as were read until that happens.
+
+        If size is -1, read as much data as possible until either the timeout
+        or EOF is encountered.
+
+        If timeout is None, there is no timeout and reading will continue until
+        either size bytes have been read, or EOF occurs. A negative timeout
+        means to use the default timeout for this process
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def read_nonblocking(self, size=1, timeout=-1):
+        """
+        This reads at most size characters from the process, or until EOF or a
+        timeout occurs. If no data is read before EOF or a timeout occurs, then
+        pexpect.EOF or pexpect.TIMEOUT are raised respectively.
+
+        A timeout of None means to wait forever for either data or EOF
```

Comment:
Why does a 'nonblocking' read support waiting for a timeout?

Reply:
Ah, this matches https://pexpect.readthedocs.io/en/stable/api/pexpect.html#pexpect.spawn.read_nonblocking